demos/mask_rcnn_demo/cpp/main.cpp

```diff
@@ -45,240 +49,327 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-        for (size_t box = 0; box < BOXES; ++box) {
-            float* box_info = do_data + box * BOX_DESCRIPTION_SIZE;
-            auto batch = static_cast<int>(box_info[0]);
-            if (batch < 0)
-                break;
-            if (batch >= static_cast<int>(netBatchSize))
-                throw std::logic_error("Invalid batch ID within detection output box");
-            float prob = box_info[2];
-            float x1 = std::min(std::max(0.0f, box_info[3] * images[batch].cols), static_cast<float>(images[batch].cols));
-            float y1 = std::min(std::max(0.0f, box_info[4] * images[batch].rows), static_cast<float>(images[batch].rows));
-            float x2 = std::min(std::max(0.0f, box_info[5] * images[batch].cols), static_cast<float>(images[batch].cols));
-            float y2 = std::min(std::max(0.0f, box_info[6] * images[batch].rows), static_cast<float>(images[batch].rows));
-            int box_width = static_cast<int>(x2 - x1);
-            int box_height = static_cast<int>(y2 - y1);
-            auto class_id = static_cast<size_t>(box_info[1] + 1e-6f);
-            if (prob > PROBABILITY_THRESHOLD && box_width > 0 && box_height > 0) {
-                size_t color_index = class_color.emplace(class_id, class_color.size()).first->second;
-                auto& color = CITYSCAPES_COLORS[color_index % arraySize(CITYSCAPES_COLORS)];
-                float* mask_arr = masks_data + box_stride * box + H * W * (class_id - 1);
-                slog::info << "Detected class " << class_id << " with probability " << prob << " from batch " << batch
-                           << ": [" << x1 << ", " << y1 << "], [" << x2 << ", " << y2 << "]" << slog::endl;
-                cv::Mat mask_mat(H, W, CV_32FC1, mask_arr);
-
-                cv::Rect roi = cv::Rect(static_cast<int>(x1), static_cast<int>(y1), box_width, box_height);
-                cv::Mat roi_input_img = output_images[batch](roi);
-                const float alpha = 0.7f;
-
-                cv::Mat resized_mask_mat(box_height, box_width, CV_32FC1);
-                cv::resize(mask_mat, resized_mask_mat, cv::Size(box_width, box_height));
-
-                cv::Mat uchar_resized_mask(box_height, box_width, CV_8UC3,
-                    cv::Scalar(color.blue(), color.green(), color.red()));
-                roi_input_img.copyTo(uchar_resized_mask, resized_mask_mat <= MASK_THRESHOLD);
-
-                cv::addWeighted(uchar_resized_mask, alpha, roi_input_img, 1.0f - alpha, 0.0f, roi_input_img);
-                cv::rectangle(output_images[batch], roi, cv::Scalar(0, 0, 1), 1);
-            }
-        }
-        metrics.update(startTime);
-        for (size_t i = 0; i < output_images.size(); i++) {
-            std::string imgName = "out" + std::to_string(i) + ".png";
-            cv::imwrite(imgName, output_images[i]);
-            slog::info << "Image " << imgName << " created!" << slog::endl;
-        }
+        size_t class_id = static_cast<size_t>(box_info[1] + 1e-6);
```

Comment:
I can see, that it was there before, but what is `+ 1e-6` for?

Reply:
I have the same question, but decide to leave this as it was in original.

Reply:
I think it's best to remove it since nobody can explain this

Reply:
my thinking is opposite, the demo has been proven to work before, so why do you want this change?

Reply:
We are responsible for this code. So it's a problem if we don't understand why it works.

Reply:
but at least it works (thanks to contributors). worse case is to do changes without understanding their consequences

Reply:
you roam the demo. It's a good moment to find out :)

Reply:
no, it is not good moment for this